python/ray/data/_internal/logical/operators.py

```diff
@@ -0,0 +1,61 @@
+import sys
+from typing import Any, Dict, Iterable, Optional, Union
+from ray.data._internal.compute import ComputeStrategy
+from ray.data._internal.lazy_block_list import LazyBlockList
+
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.data.block import BatchUDF
+from ray.data.datasource.datasource import Datasource
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+class Read(LogicalOperator):
+    """Logical operator for read."""
+
+    def __init__(
+        self,
+        blocks: LazyBlockList,
+        datasource: Datasource,
+        parallelism: int = -1,
+        ray_remote_args: Dict[str, Any] = None,
+        read_args: Dict[str, Any] = None,
+    ):
+        super().__init__("Read", [])
+        self._blocks = blocks
+        self._datasource = datasource
+        self._parallelism = parallelism
+        self._ray_remote_args = ray_remote_args
+        self._read_args = read_args
+
+
+class MapBatches(LogicalOperator):
```

Comment:
Shall we put each operator into a separate file like in the execution package?

Reply:
sure, created `operators` directory and have same structure as execution package.